controllers/finalizer.go

```diff
@@ -68,12 +69,27 @@ func addFinalizer(name string, namespace string, gvrShort string, client k8sclie
 		return nil
 	})
 
+	// get resource for event broadcasting
+	r, err2 := client.Resource(schema.GroupVersionResource{
```

Comment:
I tried to use metadata instead of querying for the real object but I could not get it to work as expected.